### PR TITLE
Convert triple_alpha_plus_cago to C++

### PR DESCRIPTION
**Pull request by maxpkatz**

### CODE REVIEW

**Comment by shardi2** [comment]
Mass_Fractions_plots_triple

**Comment by shardi2** [comment]
Mass_Fractions_plots_triple

File: networks/triple_alpha_plus_cago/actual_rhs.H [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+    t9i23 = 1.0_rt / t923;
+    t943  = std::pow(t9, (4.0_rt / 3.0_rt));
+    t9i43 = 1.0_rt / t943;
+    t932  = std::pow(t9, THREE_HALVES);
+    t9i32 = 1.0_rt / t932;
+    t953  = std::pow(t9, FIVE_THIRDS);
+    t9i53 = 1.0_rt / t953;
+    t92   = t9 * t9;
+    t9i2  = 1.0_rt / t92;
+
```
Comment (zingale) [inline networks/triple_alpha_plus_cago/actual_rhs.H:116]: it looks like t9r32, t9ri and t9ri2 are never used (in Fortran as well)

File: networks/triple_alpha_plus_cago/actual_rhs.H [hunk 2]
```diff
@@ -0,0 +1,427 @@
...
+    // triple alpha to c12 (divided by 3! = 6)
+    // from cf88
+
+    // q = -0.092;     2 He4 --> Be8
+    a    = (7.4e5_rt * t9i32) * std::exp(-1.0663_rt * t9i);
+    dadt = -a * THREE_HALVES * t9i + a * t9i2 * 1.0663_rt;
+
+    b    =  4.164e9_rt * t9i23 * exp(-13.49_rt * t9i13 - t92 / 9.604e-3_rt);
+    dbdt = -b * (2.0_rt / 3.0_rt) * t9i + b * (13.49_rt * (1.0_rt / 3.0_rt) * t9i43 - 2.0_rt * t9 / 9.604e-3_rt);
```
Comment (zingale) [inline networks/triple_alpha_plus_cago/actual_rhs.H:130]: std::exp

File: networks/triple_alpha_plus_cago/actual_rhs.H [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+    c    = 1.0_rt + 3.1e-2_rt * t913 + 8.009_rt * t923 + 1.732_rt * t9 + 49.883_rt * t943 + 27.426_rt * t953;
+    dcdt = 3.1e-2_rt * (1.0_rt / 3.0_rt) * t9i23 + 8.009_rt * (2.0_rt / 3.0_rt) * t9i13 +
+           1.732_rt + 49.883_rt * (4.0_rt / 3.0_rt) * t913 + 27.426_rt * FIVE_THIRDS * t923;
+
+    r2abe    = a + b * c;
+    dr2abedt = dadt + dbdt * c + b * dcdt;
+
+
+    // q = 7.367;      He4 + Be8 --> C12
+    a    = (130_rt * t9i32) * exp(-3.3364_rt * t9i);
```
Comment (zingale) [inline networks/triple_alpha_plus_cago/actual_rhs.H:141]: std::exp

Comment (zingale) [reply networks/triple_alpha_plus_cago/actual_rhs.H:141]: lots more of those... just check them all

**Comment by zingale** [comment]
here's the `test_rhs` comparison:
```
$ fcompare.gnu.ex react_triple_alpha_plus_cago_test_rhs.VODE.cxx react_triple_alpha_plus_cago_test_rhs.VODE.fortran

            variable name            absolute error            relative error
                                        (||A - B||)         (||A - B||/||A||)
 ----------------------------------------------------------------------------
 level = 0
 density                                          0                         0
 temperature                                      0                         0
 Ydot_helium-4                      3.637978807e-12           5.109526825e-18
 Ydot_carbon-12                     3.637978807e-12            5.10961299e-18
 Ydot_oxygen-16                     3.637978807e-12           5.109591448e-18
 Ydot_iron-56                                     0                         0
 Xold_helium-4                                    0                         0
 Xold_carbon-12                                   0                         0
 Xold_oxygen-16                                   0                         0
 Xold_iron-56                                     0                         0
 Tdot                                           480           1.547609256e-14
 Edot                                     134217728           2.728035731e-17
 J_helium-4_helium-4                1.164153218e-10           4.087518067e-18
 J_carbon-12_helium-4               1.164153218e-10           4.087724859e-18
 J_oxygen-16_helium-4               1.164153218e-10           4.087673159e-18
 J_iron-56_helium-4                               0                         0
 J_T_helium-4                                 19200           1.547596001e-14
 J_E_helium-4                            2147483648           1.091204946e-17
 J_helium-4_carbon-12               2.328306437e-10           2.780055662e-18
 J_carbon-12_carbon-12              2.328306437e-10           2.780055662e-18
 J_oxygen-16_carbon-12              2.328306437e-10           2.780055662e-18
 J_iron-56_carbon-12                              0                         0
 J_T_carbon-12                                57344           1.764003987e-14
 J_E_carbon-12                      3.435973837e+10           5.937166394e-17
 J_helium-4_oxygen-16                             0                         0
 J_carbon-12_oxygen-16                            0                         0
 J_oxygen-16_oxygen-16                            0                         0
 J_iron-56_oxygen-16                              0                         0
 J_T_oxygen-16                                    0                         0
 J_E_oxygen-16                                    0                         0
 J_helium-4_iron-56                               0                         0
 J_carbon-12_iron-56                              0                         0
 J_oxygen-16_iron-56                              0                         0
 J_iron-56_iron-56                                0                         0
 J_T_iron-56                                      0                         0
 J_E_iron-56                                      0                         0
 J_helium-4_T                       2.032879073e-20           1.719710174e-17
 J_carbon-12_T                      6.776263578e-21           5.732346506e-18
 J_oxygen-16_T                      3.388131789e-21           2.866175846e-18
 J_iron-56_T                                      0                         0
 J_T_T                              7.972121239e-07           1.548149013e-14
 J_E_T                                          0.5           6.121092746e-17
 J_helium-4_E                                     0                         0
 J_carbon-12_E                                    0                         0
 J_oxygen-16_E                                    0                         0
 J_iron-56_E                                      0                         0
 J_T_E                                            0                         0
 J_E_E                                            0                         0```

File: networks/triple_alpha_plus_cago/actual_rhs.H (hunk 2 above)
Comment (maxpkatz) [reply networks/triple_alpha_plus_cago/actual_rhs.H:130]: Fixed

**Comment by zingale** [comment]
Here's another view, integrating out longer (this is X(16O) vs t

3alpha_o12

**Pull request merged**